chore: update docs and a test related to idempotency_key (#1409)
sequinstream/sequin: docs/how-to/create-audit-logs.mdx

 Duplicates are rare and only occur if your processor successfully writes to the database but fails to acknowledge messages from the queue (SQS/Kafka) or fails to return a 200 status code (HTTP endpoints). In these cases, the message will be redelivered to ensure at-least-once delivery.

>  To protect against duplicates, each message includes a `idempotency_key` field that can be used for idempotent processing. This is particularly useful for discerning users who want to implement their own idempotency logic or for database sync scenarios where upserts are needed.

Reviewer: ```suggestion
 To protect against duplicates, each message includes a `idempotency_key` which is intended for use by your application to reject any [possible duplicate messages](https://sequinstream.com/docs/reference/sinks/overview#guaranteed-delivery).
```

>  The idempotency key is available in the message metadata as `metadata.idempotency_key`. For example, in a change message it would look like:

Reviewer: ```suggestion
 The [idempotency key](/reference/messages) is available in the message metadata as `metadata.idempotency_key`. For example, in a change message it would look like:
```

 ```json
 {
   "metadata": {
     "idempotency_key": "base64_encoded_string_here",
     ...
   }
 }
 ```

>  For regular changes (non-backfill), the idempotency key is generated by combining the transaction's position in the database log (commit_lsn) and the position within that transaction (commit_idx), then encoding it in base64. This ensures each change within a transaction has a unique identifier.

Reviewer: ```suggestion
```

>  For backfill messages, the key is generated by combining the backfill's unique identifier (backfill_id) and the primary key values of the record being processed, then encoding it in base64. This ensures each record in a backfill operation has a unique identifier, even if the same record appears multiple times.

Reviewer: ```suggestion
```
